spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/SparkCatalog.java

```diff
@@ -127,6 +128,16 @@ protected Catalog buildIcebergCatalog(String name, CaseInsensitiveStringMap opti
     optionsMap.putAll(options.asCaseSensitiveMap());
     optionsMap.put(CatalogProperties.APP_ID, SparkSession.active().sparkContext().applicationId());
     optionsMap.put(CatalogProperties.USER, SparkSession.active().sparkContext().sparkUser());
+    if (conf.get(CatalogProperties.WAREHOUSE_LOCATION) != null) {
+      optionsMap.put(
+          CatalogProperties.WAREHOUSE_LOCATION, conf.get(CatalogProperties.WAREHOUSE_LOCATION));
+    } else if (optionsMap.get(StaticSQLConf.WAREHOUSE_PATH().key()) != null) {
+      // if spark.sql.catalog.$catalogName.warehouse no setting, will try use
+      // spark.sql.warehouse.dir as catalog warehouse.
+      optionsMap.put(
+          CatalogProperties.WAREHOUSE_LOCATION,
+          optionsMap.get(StaticSQLConf.WAREHOUSE_PATH().key()));
+    }
```

Comment:
Supports `spark.sql.warehouse.dir` and assumes that `spark.sql.catalog.$catalogName.warehouse` is preferred.

I think it is more general to pass the warehouse path through `CatalogProperties.WAREHOUSE_LOCATION`? 
Because both `HadoopCatalog` and `HiveCatalog` are adapted to `CatalogProperties.WAREHOUSE_LOCATION`.

> StaticSQLConf.WAREHOUSE_PATH().key() = spark.sql.warehouse.dir

Reply:
Would be easier to read with Optional ?
```java
    final String warehouse = Optional.ofNullable(conf.get(CatalogProperties.WAREHOUSE_LOCATION))
            .orElseGet(() -> optionsMap.get(StaticSQLConf.WAREHOUSE_PATH().key()));
    if (warehouse != null) {
      optionsMap.put(CatalogProperties.WAREHOUSE_LOCATION, warehouse);
    }
```

Reply:
good hint,   I have modified.